word-guess-classes.rb

```diff
@@ -0,0 +1,235 @@
+    ]
+    @image7 = [
+      "                           __/ \\--\\".colorize(:red),
+      "                           U |_|__|".colorize(:light_red),
+      "                                ||".colorize(:yellow),
+      "                                ||".colorize(:green),
+      "                        ,   ,   ,|".colorize(:blue),
+      "                        ||__||__||".colorize(:magenta),
+      "                        ..--..--.|".colorize(:red),
+      "                        ||__||__||".colorize(:light_red),
+      "                        ..--..--.|".colorize(:yellow),
+      "||  ||  ||  ||  ||  ||  ||  ||  ||".colorize(:green)
+    ]
+    @image8 = [
+      "                           __/ \\--\\".colorize(:red),
+      "                           U |_|__|".colorize(:light_red),
+      "                                ||".colorize(:yellow),
+      "                                ||".colorize(:green),
+      "                            ,   ,|".colorize(:blue),
+      "                            ||__||".colorize(:magenta),
+      "                            ..--.|".colorize(:red),
+      "                            ||__||".colorize(:light_red),
+      "                            ..--.|".colorize(:yellow),
+      "||  ||  ||  ||  ||  ||  ||  ||  ||".colorize(:green)
+    ]
+    @gameover = [
+
+      "                                ||".colorize(:red),
+      "                                ||".colorize(:light_red),
+      "                                 |".colorize(:yellow),
+      "                                ||".colorize(:blue),
+      "                                .|".colorize(:magenta),
+      "                      __/ \\--\\  .|".colorize(:red),
+      "                      U |_|__|  .|".colorize(:light_red),
+      "||  ||  ||  ||  ||  ||  ||  ||  ||".colorize(:yellow)
+    ]
+    @images = [@image0, @image1, @image2, @image3, @image4, @image5, @image6, @image7, @image8, @gameover]
+  end
+
+  def print_image(wrong_guesses)
+    @images[wrong_guesses]
+  end
+
+end
+
+class Game
+  attr_reader :start_game, :game_play
+  def initialize(word, image)
+    @word = word
+    @image = image
+  end
+
+  def start_game
+    print "Word to Guess: "
+    @word.print_blanks
+    puts "\nYou have #{9 - @word.wrong_guesses} guesses!"
+    puts
+    puts @image.print_image(0)
+  end
+
+  def game_play
+    loop_again = true
+    until loop_again == false
+      puts "\n\nGuess a letter!"
+      print "> "
+      guess = gets.chomp.downcase
+      @word.check_guess(guess)
+      @word.print_guesses
+      puts
+      print "Word To Guess: "
+      @word.fill_blanks(guess)
+      @word.print_blanks
+      puts "\nYou have #{9 - @word.wrong_guesses} guesses left!"
+      puts @image.print_image(@word.wrong_guesses)
+      if @word.wrong_guesses == 9 || (@word.letters == @word.blanks && @word.wrong_guesses < 9)
+        if @word.letters == @word.blanks && @word.wrong_guesses < 9
```

Comment:
There's a few places through here where you use `9`, meaning the number of guesses. In programming this pattern is called a "magic number". Magic numbers are generally considered bad pattern for two reasons:

- It's not immediately obvious to the reader what 9 represents
- It will be very difficult to ever change this number - you'll have to scour your code for places where it occurs

A good alternative would be to define a constant at the top of the file, something like:

```ruby
MAX_WRONG_GUESSES = 9
```